react/testing_3/src/app/components/SimpleCalculator/index.js

```diff
@@ -0,0 +1,36 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import actionCreators from '../../../redux/simpleCalculator/actions';
+
+import SimpleCalculator from './layout';
+
+class SimpleCalculatorContainer extends Component {
+  render() {
+    return (
+      <SimpleCalculator
+        calculatorValue={this.props.calculatorValue}
+        onAdd={this.props.add}
+        onSubstract={this.props.substract}
+      />
+    );
+  }
+}
+
+const mapStateToProps = state => ({
+  calculatorValue: state.calculatorValue
+});
+
+const mapDispatchToProps = dispatch => ({
+  add: () => {
+    dispatch(actionCreators.add());
+  },
+  substract: () => {
```

Comment:
I'd use
```
const mapDispatchToProps = dispatch =>
  bindActionCreators(actionCreators, dispatch);
```
What do you think ?
https://redux.js.org/api/bindactioncreators

Reply:
I think i can use it here, in order to bind both actions to `dispatch` directly. With this we are binding our actions with our `store` instance. Now which is the real justification for this?. Because the use of `bindActionCreators` is `when you want to pass some action creators down to a component that isn't aware of Redux`.  But here we are just passing our methods `add and substract` to our `SimpleCalculator/layout.js`.
The justification is `i don't want to pass dispatch or the store` ? or `It is better in order to avoid unnecessary declarations in the mapDispatchToProps` (In the case that we don't have any extra logic when we have to dispatch any action).

What do you think?  😄

Reply:
Yeah! I agree with you!. We should start implementing this.May be in our training. What do you think?

Reply:
Agree, it's a first good step 😊